### ISSUE TITLE
Preview points are not taken into account in some cases

### ISSUE THREAD

**Issue by Morilli** [ppy/osu#3112]
If a beatmapset has multiple audio files and you switch between difficulties using different audio files in songselect while staying in that beatmapset, osu!lazer will not use the preview point to start the audio file for the newly selected difficulty, but rather start the audio from the beginning.
Video: https://streamable.com/4lzln

**Comment by Aergwyn** [comment]
Does stable restart audio if it's a different file even when staying in the same set? I can't find one to try and reproduce it so do you have a Beatmapset at hand?

#4156 addressed the part about not starting/looping from the preview point but I'm not sure if I understand the other part correctly.

**Comment by ProgrammaticNajel** [comment]
I've confirmed that stable does restart audio in correct preview points in different files even when in the same set. I tested using [this beatmapset](https://osu.ppy.sh/beatmapsets/228484). Each difficulty has a different audio file, and stable restarts the audio to their respective preview points.

(Note: I cannot provide a recording because the songs are over ~3 and a half minutes long, and one of audio tracks has long silence after it. You might need to confirm this yourself...)
